lib/active_record_shards/default_slave_patches.rb

```diff
@@ -141,8 +141,6 @@ def model.on_slave_by_default?
           left_reflection.klass.on_slave_by_default?
         end
 
-        # also transfer the sharded-ness of the left table to the join model
-        model.not_sharded unless model.left_reflection.klass.is_sharded?
```

Comment:
Will this be a problem?

Reply:
Not sure :|

Reply:
seems super sketchy that we call not_sharded ... maybe this is a 1-off join model ... which means that it should now be `model.sharded if foo` instead ?

Reply:
Can we add a test for this case?

Reply:
I've tried adding a test - it's not exactly obvious where to put it though